### ISSUE TITLE
False negative | No binding for nonlocal found

### ISSUE THREAD

**Issue by mbyrnepr2** [PyCQA/pylint#6883]
````python
def outer():
  nonlocal a
  a = 1
outer()
````

- pylint gives a success
- but if you comment out `a = 1` pylint also finds the missing binding

_Originally posted by @da-dada in https://github.com/PyCQA/pylint/issues/6812#issuecomment-1146395637_

Further information:
```python
python example.py
  File "/Users/markbyrne/programming/example.py", line 2
    nonlocal a
    ^^^^^^^^^^
SyntaxError: no binding for nonlocal 'a' found
```

**Comment by mbyrnepr2** [comment]
Created this as I thought this would deserve its own issue since it was not the original subject of #6812

**Comment by da-dada** [comment]
there is a somewhat similar behaviour with **global**
````python
A = 1
class Foo:
    global A
    A = 2

print(Foo.A)
````
I beg your pardon if I have entered this comment the wrong way again, but I simply don't find a template or anything like it you have been mentioning

**Comment by mbyrnepr2** [comment]
@da-dada can you view this template for reporting a bug? https://github.com/PyCQA/pylint/issues/new?assignees=&labels=bug%2Cneeds+triage&template=BUG-REPORT.yml

**Comment by da-dada** [comment]
@mbyrnepr2 I'll give it a try

**Comment by mbyrnepr2** [comment]
Created https://github.com/PyCQA/pylint/issues/7098 so that the current issue can stay focused on the original nonlocal issue.
